justfile

```diff
@@ -122,10 +122,36 @@ build-ingestion-service:
 
 web-dev-up:
```

Comment:
When running tests locally this can be used to start es/pg, on CI this won't work and `job.services` needs to be used.

Reply:
❓ when i run this locally, this fails with the following log. not sure if i'm missing sth in my setup? i can locally run `docker compose -f web-api/compose.db.yml up` on `main` without issues if that helps.
```
% just web-dev-up
++ command -v podman
++ command -v docker
+ ociRunner=/usr/local/bin/docker
++ command -v podman-compose
++ command -v docker-compose
+ compose=/usr/local/bin/docker-compose
++ /usr/local/bin/docker ps --format json
++ jq 'map(select(.Labels."com.docker.compose.project" == "web-api") | .Labels."com.docker.compose.service")'
+ running_services=
+ RESTART=false
+ jq -e 'contains(["ingestion"])' /dev/fd/63
++ echo ''
+ RESTART=true
+ /usr/local/bin/docker-compose -f ./web-api/compose.ingestion.yml down
ERROR: Service 'ingestion' depends on service 'elasticsearch' which is undefined.
error: Recipe `web-dev-up` failed with exit code 1
```

additionally, if i try `docker compose -f web-api/compose.db.yml up` on this branch, then i get
```
services.elasticsearch.healthcheck Additional property health-retries is not allowed
```

Reply:
it progresses further now, but still fails with the following. also it seems that the detection of running containers is not working correctly as `docker ps` shows empty results but the script still seems set `FORCE_START` to true.
```
% docker ps
CONTAINER ID   IMAGE     COMMAND   CREATED   STATUS    PORTS     NAMES

% just web-dev-up  
++ command -v podman
++ command -v docker
+ ociRunner=/usr/local/bin/docker
++ command -v podman-compose
++ command -v docker-compose
+ compose=/usr/local/bin/docker-compose
++ /usr/local/bin/docker ps --format json
++ jq 'map(select(.Labels."com.docker.compose.project" == "web-api") | .Labels."com.docker.compose.service")'
+ running_services=
+ FORCE_START=false
+ jq -e 'contains(["ingestion"]) or contains(["personalization"])' /dev/fd/63
++ echo ''
+ FORCE_START=true
+ echo 'A compose-all (i.e. composition of db and compose.ingestion.yml and/or compose.personalization.yml) is already running.'
A compose-all (i.e. composition of db and compose.ingestion.yml and/or compose.personalization.yml) is already running.
+ echo 'NOW STOPPING IT'
NOW STOPPING IT
+ /Users/janpetsche/.cargo/bin/just compose-all-down
+ :
+ [[ true != false ]]
+ /Users/janpetsche/.cargo/bin/just web-dev-down
+ :
+ rm ./web-api/assets
+ ln -s ./assets/smbert_v0003 ./web-api/assets
+ /usr/local/bin/docker-compose -f ./web-api/compose.db.yml up --detach --remove-orphans
ERROR: The Compose file '././web-api/compose.db.yml' is invalid because:
services.elasticsearch.healthcheck value 'start-period' does not match any of the regexes: '^x-'
error: Recipe `web-dev-up` failed with exit code 1
```